Lesson34/feyza/lesson34-app/src/components/Login/index.jsx

```diff
@@ -0,0 +1,38 @@
+import { useContext, useState } from "react";
+import { UserDispatchContext, UserActionTypes } from "../../userContext";
+
+export const LogIn = () => {
+  const [username, setUserName] = useState("");
+  const [password, setPassword] = useState("");
+  const { dispatch } = useContext(UserDispatchContext);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    dispatch({
+      type: UserActionTypes.login,
+      payload: { username, isLoggedInUser: true },
```

Comment:
Instead of adding `isLoggedInUser: true` here, make sure to combine it with eh incoming payload in the userReducer function instead. 'userReducer' should be the single place of truth for defining the logic. Components that dispatch actions should not need to take care of it, just provide data that they have (username in this case).